File: iphone/Classes/Device.m
/*
 *  Device.m 
 *  Used to display Device centric details handset.
 *
 *  Created by Nitobi on 12/12/08.
 *  Copyright 2008 Nitobi. All rights reserved.
 */

#import "Device.h"

@implementation Device

/**
 * returns a dictionary with various device settings
 *  - gap (version)
 *  - Device platform
 *  - Device version
 *  - Device name (e.g. user-defined name of the phone)
 *  - Device uuid
 */
- (NSDictionary*) deviceProperties
{
	UIDevice *device = [UIDevice currentDevice];
    NSMutableDictionary *devProps = [NSMutableDictionary dictionaryWithCapacity:4];
    [devProps setObject:[device model] forKey:@"platform"];
    [devProps setObject:[device systemVersion] forKey:@"version"];
    [devProps setObject:[device uniqueIdentifier] forKey:@"uuid"];
    [devProps setObject:[device name] forKey:@"name"];
    [devProps setObject:@"0.8.0" forKey:@"gap"];

    NSDictionary *devReturn = [NSDictionary dictionaryWithDictionary:devProps];
    return devReturn;
}

@end

File: javascripts/maemo/device.js
// DeviceInfo is a Qt object on Maemo

function Device() {
    this.available = true;
	this.platform  = DeviceInfo.platform;
	this.version   = DeviceInfo.version;
	this.name      = DeviceInfo.name;
	this.gap       = DeviceInfo.gap;
	this.uuid      = DeviceInfo.uuid;
    
}

navigator.Device = window.Device = window.device = new Device();
